pkg/operator/operator.go

```diff
@@ -161,22 +163,28 @@ func (o *operator) prepareConfig() error {
 	return nil
 }
 
-func (o *operator) syncComponent(ctx context.Context) func(obj interface{}) {
+func (o *operator) syncComponent(ctx context.Context, eventType operatorv1pb.ResourceEventType) func(obj interface{}) {
 	return func(obj interface{}) {
-		c, ok := obj.(*componentsapi.Component)
+		c, ok := obj.(*compapi.Component)
 		if ok {
-			log.Debugf("Observed component to be synced: %s/%s", c.Namespace, c.Name)
-			o.apiServer.OnComponentUpdated(ctx, c)
+			log.Debugf("Observed component to be synced: (%s) %s/%s", eventType, c.Namespace, c.Name)
+			o.apiServer.OnComponentUpdated(ctx, &loader.Event[compapi.Component]{
+				Resource: *c,
```

Comment:
Here seems a value copy, could we use pointer?

Reply:
We are _not_ using pointers to components or other resources extensively. I did make the change to use pointers but this created a larger diff. I agree that we should change to using pointers everywhere, but I think this should be a separate PR.

Reply:
Sure.